src/Module.Server/Rewards/CrpgRewardServer.cs

```diff
@@ -121,13 +121,15 @@ public override void OnRemoveBehavior()
     /// Update rating and statistics from the last time this method was called and also give rewards.
     /// </summary>
     /// <param name="durationRewarded">Duration for which the users should be rewarded.</param>
+    /// <param name="durationUpkeep">Duration for which the users will pay for their equipment.</param>
     /// <param name="defenderMultiplierGain">Multiplier to add to the defenders. Can be negative.</param>
     /// <param name="attackerMultiplierGain">Multiplier to add to the attackers. Can be negative.</param>
     /// <param name="valourTeamSide">Team to give valour to.</param>
     /// <param name="constantMultiplier">Multiplier that should be given to everyone disregarding any other parameters.</param>
     /// <param name="updateUserStats">True if score and rating should be saved.</param>
     public async Task UpdateCrpgUsersAsync(
         float durationRewarded,
+        float durationUpkeep,

```

Comment:
What do you think of making this null by default and use `durationRewarded` if it's null? That way you don't need to change the other game modes.
```suggestion
        float? durationUpkeep = null,
```